### ISSUE TITLE
Trying to add Tag, failed with 0 byte image or duplicate image created

### ISSUE THREAD

**Issue by kmcbest** [deckerst/aves#916]
**Describe the bug**
Add tag to an image without success, image not tagged, duplicate or 0 byte one created.

**To Reproduce** (Note: the bug isn't always reproducible)
1. Open an image, jpeg or png
2. Click on add tag feature and select some tags and click the ✓ icon
3. There are three possible results: a) the image is successfully tagged; b) the screen flashes and a black image with an "oops" is displayed, which is in fact a 0 byte image with partially the same name as the image I selected (which itself hasn't been tagged); c) the image selected isn't tagged, but a duplicate image is created with partially the same name and without tags.

**Expected behavior**
Image is tagged, as the "a)" case above.

**Screenshots**
1. The 0 byte images
![1449ce094cba04f934f12d1460d529d](https://github.com/deckerst/aves/assets/689023/16d534cc-5fad-49e6-b5ae-54312df4f720)
![34fcafbce2cc245e457e4ce25275910](https://github.com/deckerst/aves/assets/689023/ab240f6f-4bf9-4977-bfd5-df5ad18798c4)
![0d3f2ee5277df8779972f27f351435a](https://github.com/deckerst/aves/assets/689023/5e391fdb-ba11-4841-908d-b5123b22630b)
2. The duplicate images (screenshot from Total Commander, to visualize their names)
![image](https://github.com/deckerst/aves/assets/689023/d2c3d430-c3d0-4578-b446-7cd102c87c0f)

**System information and logs:**
In the app, there are instructions in the `About` page > `Bug Report` section. After following them, paste here your system information and attach your logs.
Package: deckers.thibault.aves
Installer: com.miui.packageinstaller
Aves version: 1.10.4-play, build 113
Flutter: stable 3.16.9
Android version: 14, API 34
Android build: UP1A.230905.011
Device: Xiaomi 23078RKD5C
Support: dynamic colors=true, geocoder=true, HDR=true
Mobile services: not available
System locales: zh_CN
Storage volumes: /storage/emulated/0/
Storage grants: /storage/emulated/0/Download/Hershuar/, /storage/emulated/0/APPS/, /storage/emulated/0/Download/壁纸/, /storage/emulated/0/Download/email/, /storage/emulated/0/Pictures/, /storage/emulated/0/DCIM/, /storage/emulated/0/IDMP/
Error reporting: true

[aves-logs-20240220_141913.txt](https://github.com/deckerst/aves/files/14340264/aves-logs-20240220_141913.txt)

**Additional context**
None

**Comment by deckerst** [comment]
Thanks for the report, I'll look into it.

Just to be clear, in any case (b or c) your original image is *not* deleted, right? My priority is to prevent data loss.

Do you think your storage is closed to being full when you experience the issue?

**Comment by kmcbest** [comment]
Thanks for the reply. I have a lot of free space when tagging. And no, there's no data loss, just new images (0 byte or dupes) created.
I've tried to kill and restart Aves, then restart my phone when I came across the problem, and it still exists.

I forgot to mention that, when a 0-byte or a dupe is created, if I go to the original image and tag it again, it would be a 100% success. So before I file the bug, what I did was tagging it again when the first attempt failed.
